csrc/preseg_passes/allocation_order_inference.cpp

```diff
@@ -99,4 +99,36 @@ std::unordered_map<const TensorView*, AllocationOrder> inferenceAllocationOrder(
   return alloc_order_map;
 }
 
+void AllocationDomainPass::runPass(Fusion* fusion) {
+  std::unordered_map<const TensorView*, AllocationOrder> stride_mapping =
+      inferenceAllocationOrder(fusion);
+
+  for (Val* out_val : fusion->outputs()) {
+    auto* out_tv = dynamic_cast<TensorView*>(out_val);
+    // skip:
+    //   1. non-tensor output;
+    //   2. tensor output with allocation specified, assuming everything is
+    //   semantical
+    //   3. tensor output that's aliasing (Does aliased src matter?)
+    if (out_tv == nullptr || out_tv->hasAllocation() ||
+        fusion->getOutputAlias(out_val).type != AllocationType::NoAlias) {
+      continue;
+    }
+
+    auto mapped_entry = stride_mapping.find(out_tv);
+    if (mapped_entry == stride_mapping.end() || mapped_entry->second.empty()) {
```

Comment:
What's about `mapped_entry->second.empty()`?

Reply:
(To be more verbose to reduce roundtrip) 

I was trying to see if removing this check is all fine. It looks like this only happens for rank-0 tensors and even without this check the code below is general enough to handle this special case.

Reply:
I think you are right.

I did forget to call `noReduction` on rfactor_domain of `out_tv`, which I think would make it safe for a rank-0 tensor from a reduction. Will update.